### ISSUE TITLE
[a11y] Make the Keyboard Shortcuts dialog more accessible

### ISSUE THREAD

**Issue by IsabelParedes** [jupyterlab/jupyterlab#18374]
With a focus on keyboard navigation, one of the most important elements in JupyterLab is the Keyboard Shortcuts dialog. Any user should be able to easily access this dialog and be able to navigate it with the keyboard.

### Problem

Currently, when a user opens the Shortcuts dialog, the focus moves directly to the `Close` button. This prevents the user from being able to scroll through the table with the arrow keys. A user needs to press `Shift` + `Tab` to navigate back to the table and be able to scroll.

![Video showing how a user cannot scroll the shortcuts dialog when it is first opened](https://github.com/user-attachments/assets/84ae0cfa-92e4-4c55-88bb-006bf7daba6e)

Additionally, when using a screen reader, there are no announcements regarding the contents of the dialog.

#### Make the table a proper accessible table
1. Add headers so that screen readers can announce them

| Command   | Keybinding  |
| --------- | ----------- |
| Close Tab | `Alt` + `W` |

2. Add scope attribute to those headers
3. Make the title `Keyboard Shortcuts` be the caption for the table, this is needed for screen readers.

### Additional context

The tests were performed with JupyterLab v4.5.2 and the Orca Screen Reader.

I am not a regular screen reader user, so any input from someone who is more familiar with these tools would be highly valuable.

All feedback is welcomed. Thank You!

#### Potentially relevant issues

- https://github.com/jupyterlab/jupyterlab/issues/17759
- https://github.com/jupyterlab/jupyterlab/issues/15303
- https://github.com/jupyterlab/jupyterlab/issues/15497
- https://github.com/jupyterlab/jupyterlab/issues/6573
